### PR TITLE
Speech Recognition using the Web Speech API in JavaScript

### PR DESCRIPTION
**Pull request by zolomohan**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by zolomohan** [comment]
This fixes #1077

File: articles/speech-recognition-in-javascript/index.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+let final_transcript = "";
+
+speechRecognition.onresult = (event) => {
+  let interim_transcript = "";
+};
+``` 
+
+Now let's build a string from the results array. We should run it through a loop and add the result item to the final transcript if the result item is final. If not, we should add it to the interim results string.
+
+```JavaScript
```
Comment (cyberShaw) [inline articles/speech-recognition-in-javascript/index.md:165]: These code snippets have been explained pretty well, but wouldn't it be better if we can have some helpful code comments for this block as well as for the final `speechRecognition.js` file?

**Review by cyberShaw — CHANGES_REQUESTED** [review]
Great article, @zolomohan!
Everything was explained really well and in a crisp manner. Really suited for beginners who are looking to dive into JavaScript.

Can you please look at some suggestions regarding comments in the code snippets and implement them? we can then move on to final review.

File: articles/speech-recognition-in-javascript/index.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+- A browser to view the webpage, preferably [Google Chrome](https://www.google.com/intl/en_in/chrome/) or [Firefox](https://www.mozilla.org/en-US/firefox/new/).
+
+### Cloning the Starter Code
+To focus more on speech recognition, I've prepared a starter code. You can clone it [from this repository](https://github.com/zolomohan/speech-recognition-in-javascript-starter) on GitHub. Follow the Repository's README for instructions.
+
+You can take a look at the final code in this [GitHub Repository](https://github.com/zolomohan/speech-recognition-in-javascript).
+
+In the starter code, I've set up a language and dialect select menu, two buttons to start/stop the speech recognition, and a box to display the transcript. I've used [Bootstrap](https://getbootstrap.com/) to style the webpage.
+
+![Starter Code Screen](/engineering-education/speech-recognition-using-web-speech-api-in-javascript/start_screen.png)
```
[the remark this answers is not in the document]
Comment (zolomohan) [reply articles/speech-recognition-in-javascript/index.md:38]: @cyberShaw  The Image URL is supposed to be the article's URL in the front matter. Not the Folder name.

File: articles/speech-recognition-in-javascript/index.md (hunk 1 above)
Comment (zolomohan) [reply articles/speech-recognition-in-javascript/index.md:165]: I agree. Will add them ASAP.

**Comment by zolomohan** [comment]
@cyberShaw I've added the comments to the code snippets.

File: articles/speech-recognition-in-javascript/index.md (hunk 1 above)
Comment (zolomohan) [reply articles/speech-recognition-in-javascript/index.md:163]: Done.

**Comment by cyberShaw** [comment]
> @cyberShaw I've added the comments to the code snippets.

Thank you for adding the code comments, @zolomohan!

**Review by louisefindlay23 — APPROVED** [review]
Final review complete, @hectorkambow.

**Pull request merged**